src/cpp/utils/thread_impl/thread_impl_pthread.ipp

```diff
@@ -0,0 +1,213 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <exception>
+#include <limits>
+#include <memory>
+#include <stdexcept>
+#include <system_error>
+#include <thread>
+#include <type_traits>
+#include <utility>
+
+#include <pthread.h>
+
+namespace eprosima {
+
+class thread
+{
+    // This method is a generic proxy that serves as the starting address of the thread
+    template <typename CalleeType>
+    static void* ThreadProxy(
+            void* Ptr)
+    {
+        // Take ownership of the trampoline
+        std::unique_ptr<CalleeType> Callee(static_cast<CalleeType*>(Ptr));
+        // Call the trampoline
+        (*Callee)();
+        // Finish thread
+        return nullptr;
+    }
+
+public:
+
+    using native_handle_type = pthread_t;
+    using id = pthread_t;
+
+    thread()
+        : thread_hnd_(native_handle_type())
+    {
+    }
+
+    template<class _Fn>
+    thread(
+            int32_t stack_size,
+            _Fn&& f)
+    {
+        int errnum;
+
+        // Prepare trampoline to pass to ThreadProxy
+        using CalleeType = typename std::decay<_Fn>::type;
+        std::unique_ptr<CalleeType> callee(new CalleeType(std::forward<_Fn>(f)));
+
+        // Construct the attributes object.
+        pthread_attr_t attr;
+        if ((errnum = ::pthread_attr_init(&attr)) != 0)
+        {
+            throw std::system_error(errnum, std::system_category(), "pthread_attr_init failed");
+        }
+
+        // Ensure the attributes object is destroyed
+        auto attr_deleter = [](pthread_attr_t* a)
+                {
+                    int err;
+                    if ((err = ::pthread_attr_destroy(a)) != 0)
+                    {
+                        throw std::system_error(err, std::system_category(), "pthread_attr_destroy failed");
+                    }
+                };
+        std::unique_ptr<pthread_attr_t, decltype(attr_deleter)> attr_scope_destroy(&attr, attr_deleter);
+
+        // Set the requested stack size, if given.
+        if (stack_size > 0)
```

Comment:
Our "leave default config" value is `-1`, so technically `0` should be passed as-is

Reply:
09246dfd30b4ae9db9f076a66f02fa5d0ba37b72